src/machine/predictor.h

```diff
@@ -1,23 +1,332 @@
 #ifndef PREDICTOR_H
 #define PREDICTOR_H
 
+#include "common/logging.h"
 #include "instruction.h"
 #include "memory/address.h"
+#include "predictor_types.h"
+
+#include <QObject>
+#include <QtMath>
 
 namespace machine {
 
-class Predictor {
-public:
-    virtual Address predict(Instruction inst, Address addr) = 0;
+const char *int_to_bit_string(const uint64_t value, const uint8_t size = 8);
```

Comment:
I'm not sure how the QStringView should be implemented correctly, to be honest. This implementation allows the returned value of the function to be used in all needed places without further processing.

With QStringView as return value, I need to wrap all strings in QStringViewLiteral() macro (am I using that correctly?) and convert the return value to QString, using the .toString() method, to use the value pointed at by QStringView in any of the Qt UI widgets. Additionally for the LOG() function, I need to convert the QString using qPrintable() macro to make it work withou warning or error.

For now I've implemented it like this, but please let me know the correct usage / where can I read about it.

Reply:
> With QStringView as return value, I need to wrap all strings in QStringViewLiteral() macro

Not really, just make them UTF-16 literals. `u"Hello"` Then it will work fine. Constructing the QStringView itself is cheap.

> Additionally for the LOG() function, I need to convert the QString using qPrintable() macro to make it work withou warning or error.

Yes, but that is nothing new. We are using raw C strings practically nowhere (I hope).

> convert the return value to QString, using the .toString() method

Maybe. Some places accept string view (inside qt), some do not. To be clear, I would also accept returning a QString. But not raw pointers.

Reply:
Understood. I fixed that, and will leave it with the QStringView as a return value, if that is preferred.